### PR TITLE
Subtract areas to be avoided from ship stats

### PR DESCRIPTION
**Pull request by crlang44**
Created the following new fields which integrate the subtraction of nm in the atba from the total nm under 10 knots

ais_vsr_segments:
- is_in_atba

ship_stats_daily:
- coop_score_daily_with_atba_subtracted
- coop_grade_daily_with_atba_subtracted
- rolling_coop_score_with_atba_subtracted
- rolling_coop_grade_with_atba_subtracted
- total_distance_nm_in_atba
- cum_sum_total_distance_nm_in_atba

ship_stats_monthly:
- coop_score_monthly_with_atba_subtracted
- month_grade_with_atba_subtracted
- rolling_coop_score_with_atba_subtracted
- rolling_month_grade_with_atba_subtracted
- total_distance_nm_in_atba

ship_stats_annual:
- coop_score_annual_with_atba_subtracted
- year_grade_with_atba_subtracted
- rolling_coop_score_with_atba_subtracted
- total_distance_nm_in_atba
- cum_sum_total_distance_nm_in_atba

operator_stats_daily:
- coop_score_daily_with_atba_subtracted
- day_grade_with_atba_subtracted
- rolling_coop_score_with_atba_subtracted
- rolling_day_grade_with_atba_subtracted
- total_distance_nm_in_atba
- cum_sum_total_distance_nm_in_atba

operator_stats_montly:
- coop_score_monthly_with_atba_subtracted
- cooperation_score_monthly_with_atba_subtracted
- month_grade_with_atba_subtracted
- rolling_coop_score_with_atba_subtracted
- rolling_month_grade_with_atba_subtracted
- total_distance_nm_in_atba
- cum_sum_total_distance_nm_in_atba

operator_stats_annual:
- coop_score_annual_with_atba_subtracted
- cooperation_score_annual_with_atba_subtracted
- year_grade_with_atba_subtracted
- rolling_coop_score_with_atba_subtracted
- total_distance_nm_in_atba
- cum_sum_total_distance_nm_in_atba

### CODE REVIEW

File: sql_v3_scheduled-on-BigQuery/update_whalesafe_v3_ais_vsr_segments.sql [hunk 1]
```diff
@@ -36,20 +36,88 @@ CREATE TEMPORARY TABLE `temp_ais_vsr_segments` (
...
+  s.*, 
+  z.vsr_category, 
+  z.region AS vsr_region, 
+  -- # Spatial intersection case statement, when linestring is covered by VSR zone, return linestring
+  -- # Otherwise, intersect the linestring with the VSR zone polygon.
+  CASE WHEN ST_CoveredBy (s.linestring, z.geog) THEN s.linestring ELSE ST_Intersection(s.linestring, z.geog) END AS vsr_linestring,
+  TRUE AS is_in_atba
+FROM
+  `whalesafe_v3.ais_segments` AS s 
+  INNER JOIN `whalesafe_v3.vsr_zones` AS z ON ST_Intersects(s.linestring, z.geog) AND LOWER(z.type) = 'vsr'
```
Comment (crlang44) [inline sql_v3_scheduled-on-BigQuery/update_whalesafe_v3_ais_vsr_segments.sql:73]: Make sure to match on VSR category

File: sql_v3_scheduled-on-BigQuery/update_whalesafe_v3_ship_stats_annual.sql [hunk 2]
```diff
@@ -87,7 +79,43 @@ SELECT
...
+      ROUND(
+        GREATEST(0,
+          SUM(total_distance_nm_under_10) OVER (
+            PARTITION BY mmsi, 
+            vsr_region 
+            ORDER BY 
+              mmsi, 
+              year ROWS BETWEEN 12 PRECEDING 
+              AND CURRENT ROW
+          ) - SUM(total_distance_nm_in_atba) OVER (
```
Comment (crlang44) [inline sql_v3_scheduled-on-BigQuery/update_whalesafe_v3_ship_stats_annual.sql:93]: Update to be consistent with other parts, move the subtraction inside the sum

File: sql_v3_scheduled-on-BigQuery/update_whalesafe_v3_ship_stats_monthly.sql [hunk 3]
```diff
@@ -64,153 +85,93 @@ FROM
...
-              ORDER BY 
-                mmsi, 
-                month, 
-                year ROWS BETWEEN 12 PRECEDING 
-                AND CURRENT ROW
+            GREATEST(0,
+              (ROUND(SUM(total_distance_nm_under_10) OVER (
+                PARTITION BY mmsi, vsr_region, year ORDER BY mmsi, month, year ROWS BETWEEN 12 PRECEDING AND CURRENT ROW
+              ), 2)
+              - ROUND(SUM(total_distance_nm_in_atba) OVER (
```
Comment (crlang44) [inline sql_v3_scheduled-on-BigQuery/update_whalesafe_v3_ship_stats_monthly.sql:139]: Update to be consistent with other parts, move the subtraction inside the sum

**Review by ramrhodes — APPROVED** [review]
Ship it

**Comment by crlang44** [comment]
Deployed all to scheduled queries

**Pull request merged**